### PR TITLE
chore: Add GH action and release please support

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: .github/actions/build/action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  steps:
+    - name: Build distribution files
+      shell: bash
+      run: poetry build
+    - name: Hash build files for provenance 
+      id: package-hashes
+      shell: bash 
+      working-directory: ./dist
+      run: |
+        echo "package-hashes=$(sha256sum * | base64 -w0)" >> "$GITHUB_OUTPUT"
```
Comment (cwaldren-ld) [inline .github/actions/build/action.yml:19]: newline

File: .github/workflows/manual-publish.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  release-provenance:
+    needs: [ 'build-publish' ]
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.10.0
+    with:
+      base64-subjects: "${{ needs.build-publish.outputs.package-hashes }}"
```
Comment (cwaldren-ld) [inline .github/workflows/manual-publish.yml:51]: Curious why this is quoted? is it an array or something interesting?

File: .github/workflows/release-please.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    permissions:
+      id-token: write # Needed if using OIDC to get release secrets.
+      contents: write # Contents and pull-requests are for release-please to make releases.
+      pull-requests: write
+    outputs:
+      release-created: ${{ steps.release.outputs.release_created }}
+      upload-tag-name: ${{ steps.release.outputs.tag_name }}
+      package-hashes: ${{ steps.build.outputs.package-hashes}}
+    steps:
+      - uses: google-github-actions/release-please-action@v3
```
Comment (cwaldren-ld) [inline .github/workflows/release-please.yml:19]: Would using @v4 work?

**Review by cwaldren-ld — APPROVED** [review]

File: .github/workflows/manual-publish.yml (hunk 2 above)
Comment (keelerm84) [reply .github/workflows/manual-publish.yml:51]: I'm not sure. Rouel did the original provenance work. I've just been applying his templates to my other repos. My guess is just protecting from some weird characters or just being overly cautious.

File: .github/workflows/release-please.yml (hunk 3 above)
Comment (keelerm84) [reply .github/workflows/release-please.yml:19]: Maybe. Ryan had a similar question on the ruby otel branch. I'm just keeping my repos all using the same version of release please for now, and then I'll go through and update them in mass. This way it's a one and done thing. It also let's this be a copy-paste job, so I'm confident the first release will work.

**Pull request merged**